pageserver/client/src/page_service.rs

```diff
@@ -108,9 +108,32 @@ pub struct RelTagBlockNo {
 }
 
 impl PagestreamClient {
-    pub async fn shutdown(mut self) {
-        let _ = self.cancel_on_client_drop.take();
-        self.conn_task.await.unwrap();
+    pub async fn shutdown(self) {
```

Comment:
I wonder if we should `impl Drop` instead of having a manual shutdown function. Also okay with a manual shutdown, but would love to hear points in favour/against.

Reply:
Doesn't work because no async Drop